infrastructure/simulation-tool/src/utils/wallet.utils.ts

```diff
@@ -0,0 +1,67 @@
+import {
+    SyncProvider as RollupProvider,
+    Transaction,
+    Wallet as RollupWallet,
+    RestProvider
+} from '@rsksmart/rif-rollup-js-sdk';
+import { Wallet as EthersWallet, providers } from 'ethers';
+import config from './config.utils';
+import { NETWORK_TO_DERIVATION_PATH, CHAIN_TO_NETWORK, HardenedBit } from '../constants/network';
+
+type RollupWalletGenerator = AsyncGenerator<RollupWallet, RollupWallet>;
+
+const baseDerivationPath: `m/44'/${number}'` = NETWORK_TO_DERIVATION_PATH[CHAIN_TO_NETWORK[config.chainId]];
+
+export function getAccountPath(index: number): `m/44'/${number}'/${number}'/0/0` {
+    if (typeof index !== 'number' || index < 0 || index >= HardenedBit || index % 1) {
+        throw Error(`invalid account index: ${index}`);
+    }
```

Comment:
I'd avoid this check since it's already performed by [the library to validate the path](https://github.com/ethers-io/ethers.js/blob/0bfa7f497dc5793b66df7adfb42c6b846c51d794/packages/hdnode/src.ts/index.ts#L223).

Reply:
this is not done by the library, because the library does it on the 'address' derivation index, whereas we do it on the 'account' index, to get different accounts, rather than different addresses for the same account.

Reply:
changed